mapbox_tilesets/scripts/cli.py

```diff
@@ -513,19 +516,42 @@ def add_source(ctx, username, id, features, no_validation, token=None, indent=No
         for feature in features:
             if not no_validation:
                 utils.validate_geojson(feature)
+
             file.write((json.dumps(feature) + "\n").encode("utf-8"))
 
         file.seek(0)
+
         m = MultipartEncoder(fields={"file": ("file", file)})
-        resp = s.post(
-            url,
-            data=m,
-            headers={
-                "Content-Disposition": "multipart/form-data",
-                "Content-type": m.content_type,
-            },
+        prog = click.progressbar(
+            length=m.len, fill_char="=", width=0, label="upload progress"
```

Comment:
Maybe hoist this `prog` definition into the `else` statement?

Reply:
hoisted! and moved the `MultipartEncoder` definitions next to each post request. https://github.com/mapbox/tilesets-cli/pull/64/commits/4a56ce6afde3b1a57ebcbd1c6c65a495b1bb5c58